skyportal/plot.py

```diff
@@ -547,57 +567,684 @@ def make_custom_buttons_div(panel_name):
+    period,
+    grouped_data,
+    plot,
+    model_dict,
+    device,
+    width,
+    layout,
+):
+    """Makes period controls to be used in a period photometry panel.
+    Returns
+    -------
+    bokeh column object
+    """
+    period_selection = RadioGroup(labels=period_labels, active=0)
+
+    phase_selection = RadioGroup(labels=["One phase", "Two phases"], active=1)
+    period_title = Div(text="Period (days): ")
+    period_textinput = TextInput(value=str(period if period is not None else 0.0))
+    period_textinput.js_on_change(
+        'value',
+        CustomJS(
+            args={
+                'textinput': period_textinput,
+                'numphases': phase_selection,
+                'n_labels': len(grouped_data),
+                'p': plot,
+                **model_dict,
+            },
+            code=open(
+                os.path.join(
+                    os.path.dirname(__file__), '../static/js/plotjs', 'foldphase.js'
+                )
+            ).read(),
+        ),
+    )
+    # a way to modify the period
+    period_double_button = Button(label="*2", width=30)
+    period_double_button.js_on_click(
+        CustomJS(
+            args={'textinput': period_textinput},
+            code="""
+            const period = parseFloat(textinput.value);
+            textinput.value = parseFloat(2.*period).toFixed(9);
+            """,
+        )
+    )
+    period_halve_button = Button(label="/2", width=30)
+    period_halve_button.js_on_click(
+        CustomJS(
+            args={'textinput': period_textinput},
+            code="""
+                    const period = parseFloat(textinput.value);
+                    textinput.value = parseFloat(period/2.).toFixed(9);
+                    """,
+        )
+    )
+    # a way to select the period
+    period_selection.js_on_click(
+        CustomJS(
+            args={'textinput': period_textinput, 'periods': period_list},
+            code="""
+            textinput.value = parseFloat(periods[this.active]).toFixed(9);
+            """,
+        )
+    )
+    phase_selection.js_on_click(
+        CustomJS(
+            args={
+                'textinput': period_textinput,
+                'numphases': phase_selection,
+                'n_labels': len(grouped_data),
+                'p': plot,
+                **model_dict,
+            },
+            code=open(
+                os.path.join(
+                    os.path.dirname(__file__), '../static/js/plotjs', 'foldphase.js'
+                )
+            ).read(),
+        )
+    )
+    if device == "mobile_portrait":
+        period_controls = column(
+            row(
+                period_title,
+                period_textinput,
+                period_double_button,
+                period_halve_button,
+                width=width,
+                sizing_mode="scale_width",
+            ),
+            phase_selection,
+            period_selection,
+            width=width,
+        )
+    else:
```

Comment:
I would make all these cases where it needs to be one of "flux", "mag" or "period" explicit `elif` for all three options, and add an `else` with a `ValueError`. 

Even though these are internal functions and it is very unlikely that the wrong key would ever be used, it makes it clear that the last `elif` is for "period" and not something generic.

Reply:
Good point, updated that.